### DISCUSSION TITLE
PnP search results are worse than native - how can i get the same results as native results?

### DISCUSSION THREAD

**Discussion by steveh2023** [microsoft-search/pnp-modern-search#3684] [Q&A] [answered]
Hi,

PnP search results are worse than native - how can i get the same results as native results?

On my search page i have the pnp search box, pnp search results and pnp filters all configured, all working.

However if i search using keyword: patient for example my results are not as relevant to this keyword as if i just use the default sharepoint search box - top middle of most sites.

I want my pnp search results to be as good as the results shown when the default search box is used.

I have checked all the settings and just don't know what is wrong - I inherited this from a developer who has left and it all looks ok to me in the config. but something must be wrong - not even sure how to investigate really.

Can someone please suggest some things to try?

Thanks!

**Comment by kasperbolarsen** [comment, marked as answer]
Let me guess, you are using SharePoint as the data source in the results web part? 
That would explain the difference in the results and the search box in the topbar is using Microsoft search. 
Those two search engines are ranking the results based on their own ranking models, so if you want the PnP to be similar,  you will have to switch to the Microsoft search as a source. 
Please note that I used the term similar,  not identical.  Microsoft is using Microsoft search AND some internal tricks to provide those results. 
In general we don't recommend using PnP Modern search as a replacement for out of the box MS search, but rather to use PnP for specific workloads where MS search just doesn't deliver the capabilities we need. 
https://www.techmikael.com/2020/09/microsoft-search-vs-pnp-modern-search.html?m=1

**Comment by steveh2023** [comment]
Hi,

Yes it is using SharePoint as the data source...thanks I might switch it over next week.

**Reply by wobba** [reply]
It's also impossible to get the exact same results as out of the box due to how Microsoft handles search internally. But as @kasperbolarsen  points out, using Graph API should make it closer.
